From 22a04c524399c85f3d7b7d20a91abdde05e4cf3a Mon Sep 17 00:00:00 2001
From: Kiran Kumar K <kirankumark@marvell.com>
Date: Tue, 5 Oct 2021 10:50:40 +0530
Subject: [PATCH 06/11] octeontx2-af: Fix issue with CPT HDR parsing

commit 34eda3eaed92fa4a8ea5a28590f55164fd3022f0 from
git@git.assembla.com:cavium/WindRiver.linux.git

Packet received from 2nd pass, has CPT header prepended in big endian
format, and format cannot be changed. So, changing the variable length
offset to byte 0 instead of checking at byte 7.

Fixes: e93310e5e7fcc ("octeontx2-af: Adjust LA pointer for cpt parse header")

Signed-off-by: Kiran Kumar K <kirankumark@marvell.com>
Change-Id: Ia946144613935a015d482a9982ba93fac2efba01
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/62732
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
index e16203c39cf3..5ff30b6af5d4 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
@@ -1013,7 +1013,7 @@ static struct npc_kpu_profile_action ikpu_action_entries[] = {
 		NPC_S_KPU1_CPT_HDR, 40, 0,
 		NPC_LID_LA, NPC_LT_NA,
 		0,
-		7, 7, 0, 0,
+		0, 7, 0, 0,
 
 	},
 	{
-- 
2.31.1

